texar/torch/modules/networks/networks.py

```diff
@@ -87,12 +87,9 @@ def default_hparams():
 
     @property
     def output_size(self) -> int:
-        r"""The final dimension(s) of :meth:`forward` output tensor(s).
-
-        Here final dimension equals to ``1`` if unable to get output size
-        from network layers.
+        r"""The feature size of network layers result.
+        If unable to get output size from network layers, equals to ``1``.
         """
-
         for layer in reversed(self._layers):
             size = get_output_size(layer)
```

Comment:
I'm not sure if this implementation is correct though. You could have arbitrary layers here, and receiving a size of `None` doesn't mean the layer output size is the same as its input size -- it just means we don't know how to compute it.

A better approach would be to make `get_output_size` work for more cases. For instance, we can also create a list of modules that does not alter the tensor size, among them activations and dropouts. We also need to check if the layer has `output_size` as a property, and if it does, return that property instead.

Reply:
I see, how should we use the list of modules that does not alter the tensor size in computing `output_size`? Thanks!

Reply:
I may see what you mean, thanks!